cs/playground/FasterLogSample/Program.cs

```diff
@@ -87,9 +87,9 @@ static void ScanThread()
             {
                 while (true)
                 {
-                    while (!iter.GetNext(out result))
+                    while (!iter.GetNext(out result, out int length))
                     {
-                        Thread.Sleep(1000);
+                        iter.WaitAsync().GetAwaiter().GetResult();
```

Comment:
I wonder if this should be an `IAsyncEnumerable<T>` instead? Can change later, but it is likely a nicer abstraction for log scanning

Reply:
+1! Will that need a .net core 3.0 dependency, though? I see that right now FasterLog targets netstandard20.

Reply:
We don't need .NET Core 3.0 for `IAsyncEnumerable<T>`. Orleans uses it and we target netstandard2.0, too. You do need to reference newer libraries, though and you don't get `yield` support AFAIK unless you update to the .NET Core 3.0 *SDK* (possibly bumping to `<LangVersion>8.0</LangVersion>` in the csproj)

Reply:
The problem is that T in our case would be Span<byte>, and we cannot have a Span as a generic parameter.

Reply:
And our iterator returns two things: a span of byte, and an actual length of the item being returned. This allows the user to provide us with a potentially larger span (when we call GetMemory), if they are pooling.